### PR TITLE
raise (Failure ..) -> failwith ..

### PR DESCRIPTION
**Pull request by rgrinberg**
simple tweak

### CODE REVIEW

**Comment by rgrinberg** [comment]
The travis failure is unrelated and should be ignored.

File: async/cohttp_async.ml [hunk 1]
```diff
@@ -25,14 +25,14 @@ module Net = struct
...
-    | None -> raise (Failure "Net.connect") (* TODO proper exception *)
+    | None -> failwith "Net.connect" (* TODO proper exception *)
     | Some port ->
       let open Unix in
       Addr_info.get ~host [Addr_info.AI_FAMILY PF_INET; Addr_info.AI_SOCKTYPE SOCK_STREAM]
       >>= function
       | {Addr_info.ai_addr=ADDR_INET(addr,_)}::_ ->
         return (host, Ipaddr_unix.of_inet_addr addr, port)
-      | _ -> raise (Failure "resolution failed")
+      | _ -> failwith "resolution failed"
```
Comment (dbuenzli) [inline async/cohttp_async.ml]: I'd rather say you are using exceptions for something unexceptional.

Comment (rgrinberg) [inline async/cohttp_async.ml]: Yes all of this stuff is very much less than ideal. I think we are saving the refactoring post 1.0 however.

Comment (hannesm) [inline async/cohttp_async.ml]: why post 1.0? shouldn't such API changes (I suspect `raise` will be transformed to use an `'a option` return type instead) be before 1.0?

Comment (rgrinberg) [inline async/cohttp_async.ml]: Well we did ask people to participate in making a checklist for 1.0:

https://github.com/mirage/ocaml-cohttp/issues/198

and the issue wasn't included. I guess it's not too late now :P

Comment (rgrinberg) [inline async/cohttp_async.ml]: @hannesm @dbuenzli #247 addresses this

**Pull request merged**